src/main/java/com/o19s/es/ltr/ranker/ranklib/learning/CoorAscent.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright [2017] Wikimedia Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*===============================================================================
+ * Copyright (c) 2010-2012 University of Massachusetts.  All Rights Reserved.
+ *
+ * Use of the RankLib package is subject to the terms of the software license set 
+ * forth in the LICENSE file included with this software, and also available at
+ * http://people.cs.umass.edu/~vdang/ranklib_license.html
+ *===============================================================================
+ */
+
+package com.o19s.es.ltr.ranker.ranklib.learning;
+
+import com.o19s.es.ltr.feature.FeatureSet;
+import com.o19s.es.ltr.ranker.ranklib.metric.MetricScorer;
+import com.o19s.es.ltr.ranker.ranklib.utils.KeyValuePair;
+import com.o19s.es.ltr.ranker.ranklib.utils.MergeSorter;
+import org.elasticsearch.common.io.FastStringReader;
+
+import java.io.BufferedReader;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * @author vdang
+ * 
+ * This class implements the linear ranking model known as Coordinate Ascent. It was proposed in this paper:
+ *  D. Metzler and W.B. Croft. Linear feature-based models for information retrieval. Information Retrieval, 10(3): 257-274, 2007.
+ */
+public class CoorAscent extends Ranker {
```

Comment:
This should be unnecessary, at ranking time coordinate ascent is just a weighted sum like the current linear model